docs/types.rst

```diff
@@ -60,15 +60,14 @@ operators are :ref:`literals<rational_literals>` (or literal expressions).
 Division by zero and modulus with zero throws a runtime exception.
 
 The result of a shift operation is the type of the left operand. The
-expression ``x << y`` is equivalent to ``x * 2**y``, and ``x >> y`` is
-equivalent to ``x / 2**y``. This means that shifting negative numbers
-sign extends. Shifting by a negative amount throws a runtime exception.
+expression ``x << y`` is equivalent to ``x * 2**y``, and, for positive integers,
+``x >> y`` is equivalent to ``x / 2**y``. For negative ``x``, ``x >> y``
+is equivalent to dividing by a power of ``2`` while rounding down (towards negative infinity).
```

Comment:
We should probably say something about literals. `-1>>2` will result in `-1/2` and thus `int(-1>>1)` will be different from `int(-1) >> 1`, right?

Reply:
There is also a comment in Types.cpp:1057 (BinaryOperatorResult of RationalNumberType).

Reply:
I hadn't thought about literals actually - I think they need to be reworked as well.
Looking at Types.cpp:1060-1077 it looks like ``-1>>1`` will actually perform logical shift and not arithmetic shift and result in 0 right now. Similarly ``-1>>2``, etc.

The question is: *should* ``-1>>1`` be ``-1/2`` or should it be ``-1``?

Reply:
It should be consistent with `int(-1) >> 1` I would say. @axic what do you think?

Reply:
I would agree that ``-1>>1`` should be consistent with ``int(-1) >> 1`` and therefore result in ``-1``. Fractional operands are disallowed anyways and one can always use ``-1/(2^1)`` (resp. ``-1/(2^c)`` instead of ``-1>>c``, where ``c`` is a positive literal) instead, if one wants fractional results.

In any case I will add test cases for right shifting negative literals - they were apparently missing or haven't been updated.